### PR TITLE
Use webpack-dev-server in yarn watch/dev

### PR DESCRIPTION
**Pull request by xPaw**
This is from Vue branch, enables hot reloading.

### CODE REVIEW

**Comment by astorije** [comment]
To make sure I got this right: the only thing it provides (at least for now) is that page auto-reloads on change as opposed to previous `yarn watch` where you had to hit refresh by hand, correct?

File: package.json [hunk 1]
```diff
@@ -22,7 +22,7 @@
     "test:browser": "webpack-dev-server --config=webpack.config-browser.js",
     "test:client": "nyc --nycrc-path=test/.nycrc-mocha-webpack mocha-webpack --colors --opts=test/mocha-webpack.opts",
     "test:server": "nyc --nycrc-path=test/.nycrc-mocha mocha --colors",
-    "watch": "webpack --watch"
+    "watch": "webpack-dev-server --hot --inline"
```
Comment (astorije) [inline package.json]: Do you actually need to specify `--inline`?

```
  --inline      Inline mode (set to false to disable including client scripts
                like livereload)                       [boolean] [default: true]
```

Comment (astorije) [reply package.json]: Does `--hot` bring anything at the moment? (I couldn't tell the difference with/without)

**Comment by astorije** [comment]
@xPaw do you need to do any further changes on this?

**Comment by williamboman** [comment]
Worth noting is that `webpack-dev-server` has sort of been discontinued in favor for https://github.com/webpack-contrib/webpack-serve.

**Comment by xPaw** [comment]
Damn it, webpack

**Pull request closed without merging**